### ISSUE TITLE
crates.io-index lists dependencies incorrectly/incompletely

### ISSUE THREAD

**Issue by 51328f10** [rust-lang/crates.io#3804]
**Describe the bug**
Some crates (e.g. `rand`) require dependencies using the `package` key, so the key in the table is different from the actual crate pulled. The index lists the table key, not the crate ID. In `rand`'s case, this means tooling tries to pull `packed_simd:0.3.5`, which doesn't exist and wouldn't satisfy the dependency if it did, instead of `packed_simd_2:0.3.5`, which does and would.

**To Reproduce**
1. Go to [`rand`'s Cargo.toml](https://github.com/rust-random/rand/blob/81b8a7542b83313d6464d694ac3f86f4b0a8c084/Cargo.toml#L63) or [the web API](https://crates.io/api/v1/crates/rand/0.8.4/dependencies) or clone it and `cargo build` it, watching the dependencies it builds.
2. Notice the crate which is *actually pulled* to satisfy the dependency is `packed_simd_2`, as defined in the `package` key.
3. Look at [crates.io-index's entry for rand 0.8.4](https://github.com/rust-lang/crates.io-index/blob/master/ra/nd/rand#L67); specifically, the `deps[].name` keys. `jq` makes it easier to read: `tail -n1 crates.io-index/ra/nd/rand | jq '.deps[].name'`.
4. Notice that lists `packed_simd` instead.
5. Notice there's no other indication of what crate is being depended on.

**Expected behavior**
I expect either `name` to indicate the actual, depended-on crate, **or** for there to be another key in there telling me what the actual crate is. The web API does the latter; for the git repo it may be better to do the former for size reasons, as long as there isn't a compelling reason to keep the table key around.

I'd also *very much* like if `name` was changed to something that sounded less relevant, but as you can probably tell, I don't really know what Cargo calls anything.

**Desktop (please complete the following information):**
- All of them (I hope??)

**Smartphone (please complete the following information):**
- All of them (I hope??)

**Comment by 51328f10** [comment]
This seems like it'd be an easy fix, just doing `dependency.package.unwrap_or(key)` or something like that, but I can't actually find the code producing `crates.io-index`. If someone could point me that way I'd be happy to pitch in a PR.

**Comment by 51328f10** [comment]
So it turns out it's there (there's a `package` key), which is... weird. Before I wrote this issue I did, copy-pasting from my shell history, `grep packed_simd_2 ra/nd/rand`, and it didn't give anything. I dug through it with JQ, and didn't think to save the results, but I _definitely_ didn't see `packed_simd_2`; I used grep to check that too.

I fully deleted the repo and re-cloned, and now everything works as expected, so I'm closing this issue, because my computer's apparently haunted.
